cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -286,6 +286,145 @@ __inline__ __device__ T parse_numeric(const char* begin,
+ */
+__device__ __inline__ void trim_zeros(char const*& raw_data, long& digit_count)
+{
+  char const* start_ptr = raw_data;
+  while (*raw_data == '0' && (raw_data < start_ptr + digit_count)) { raw_data++; }
+  digit_count -= raw_data - start_ptr;
+}
+
+/**
+ * @brief Lexicographically compare digits in input against int64 max
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @return bool True if integer represented by character string is less
+ * than or equal to int64 max
+ */
+__device__ __inline__ bool lex_compare_int64max(const char* raw_data)
+{
+  constexpr int int64_max_len = 19;
+  const char int64_max[]      = {
+    '9', '2', '2', '3', '3', '7', '2', '0', '3', '6', '8', '5', '4', '7', '7', '5', '8', '0', '7'};
+  for (int i = 0; i < int64_max_len; ++i) {
+    if (raw_data[i] < int64_max[i]) {
+      return true;
+    } else if (raw_data[i] > int64_max[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * @brief Lexicographically compare digits in input against int64 min
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @return bool True if integer represented by character string is greater
+ * than or equal to int64 min
+ */
+__device__ __inline__ bool lex_compare_int64min(const char* raw_data)
+{
+  constexpr int int64_min_len = 19;
+  const char int64_min[]      = {
+    '9', '2', '2', '3', '3', '7', '2', '0', '3', '6', '8', '5', '4', '7', '7', '5', '8', '0', '8'};
+  for (int i = 0; i < int64_min_len; ++i) {
+    if (raw_data[i] < int64_min[i]) {
+      return true;
+    } else if (raw_data[i] > int64_min[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * @brief Lexicographically compare digits in input against uint64 max
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @return bool True if integer represented by character string is less
+ * than or equal to uint64 max
+ */
+__device__ __inline__ bool lex_compare_uint64max(const char* raw_data)
+{
+  constexpr int uint64_max_len = 20;
+  const char uint64_max[]      = {'1', '8', '4', '4', '6', '7', '4', '4', '0', '7',
+                             '3', '7', '0', '9', '5', '5', '1', '6', '1', '5'};
+  // This function is only called on raw_data such that the number of digits in
+  // raw_data is 20.
+  for (int i = 0; i < uint64_max_len; ++i) {
+    if (raw_data[i] < uint64_max[i]) {
+      return true;
+    } else if (raw_data[i] > uint64_max[i]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * @brief Determine which counter to increment when a sequence of digits
+ * and a parity sign is encountered.
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @param digit_count Total number of digits
+ * @param stats Reference to structure with counters
+ * @return cudf::size_type* Pointer to appropriate counter that belong to
+ * the interpreted data type
+ */
+__device__ __inline__ cudf::size_type* get_counter_address(char const* raw_data,
```

Comment:
Don't like the name... maybe something like `infer_integral_field_counter`? 
Also, `digit_count` should really be field length, given that we could have non-digit characters in there.